src/solid_poro_3D_ele/4C_solid_poro_3D_ele_pressure_velocity_based_evaluate.cpp

```diff
@@ -7,13 +7,58 @@
 
 #include "4C_fem_general_cell_type.hpp"
 #include "4C_fem_general_cell_type_traits.hpp"
+#include "4C_fem_general_element.hpp"
 #include "4C_fem_general_utils_interpolation.hpp"
 #include "4C_mat_structporo.hpp"
 #include "4C_solid_3D_ele_calc_lib.hpp"
+#include "4C_solid_3D_ele_factory.hpp"
 #include "4C_solid_poro_3D_ele_pressure_velocity_based.hpp"
+#include "4C_solid_scatra_3D_ele_factory.hpp"
+#include "4C_utils_exceptions.hpp"
+
+#include <type_traits>
+#include <variant>
 
 FOUR_C_NAMESPACE_OPEN
 
+namespace
+{
+  template <typename T, typename Variant>
+  struct IsMemberOfVariant;
+
+  template <typename T, typename... Types>
+  struct IsMemberOfVariant<T, std::variant<Types...>>
+      : public std::disjunction<std::is_same<T, Types>...>
+  {
+  };
+
+  template <typename T>
+  concept IsSolidInterface = IsMemberOfVariant<T, Discret::Elements::SolidCalcVariant>::value;
+
+  template <typename T>
+  concept IsSolidScatraInterface =
+      IsMemberOfVariant<T, Discret::Elements::SolidScatraCalcVariant>::value;
+
+  template <typename SolidInterface>
+  const auto& get_location_array(const Core::Elements::LocationArray& la)
+  {
+    if constexpr (IsSolidInterface<SolidInterface>)
+    {
+      // The solid interface only expects the first location array
+      return la[0].lm_;
+    }
+    else if constexpr (IsSolidScatraInterface<SolidInterface>)
+    {
+      // The solid-scatra interface expects all location arrays!
+      return la;
+    }
+
+    FOUR_C_THROW(
+        "Unknown type of solid variant! Either expecting a normal solid interface or a "
+        "solid-scatra interface!");
```

Comment:
```suggestion
else
{
  []<bool always_false = false>() {static_assert(always_false, "Normal or solid-scatra interface required."): }
}
```

Alternative: specialize `get_location_array()` for the two concepts `IsSolidInterface` and `IsSolidScatraInterface`?